portfolio/src/main/java/com/google/sps/servlets/SentimentAnalysis.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet which processes request to analyze message. */
+@WebServlet("/sentiment")
+public class SentimentAnalysis extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String message;
+    
+    // Check for the message parameter in request.
+    try {
+      message = request.getParameter("message");
+    } catch(Exception e) {
+      message = "";
+    }
+
+    if (message == "") {
+      response.setContentType("text/html;");
```

Comment:
It is better to return structured JSON, rather than lain text. You could create a class with two fields: a score (float) and an error message (string) and send that back as JSON. Or you could just have one field (the score) and handle the check that at least one word is entered in the javascript on the client and on the server just return 0 if the string is empty.